### PR TITLE
*: update context pkg

### PR DESCRIPTION
**Pull request by xiang90**
Fix memory leak in context pkg.

### CODE REVIEW

File: Godeps/_workspace/src/golang.org/x/net/context/withtimeout_test.go [hunk 1]
```diff
@@ -8,7 +8,7 @@ import (
 	"fmt"
 	"time"
 
-	"github.com/coreos/etcd/Godeps/_workspace/src/golang.org/x/net/context"
+	"golang.org/x/net/context"
```
Comment (yichengq) [inline Godeps/_workspace/src/golang.org/x/net/context/withtimeout_test.go]: Could it use godep's context?

Comment (xiang90) [inline Godeps/_workspace/src/golang.org/x/net/context/withtimeout_test.go]: no idea. but i fixed it.

**Comment by xiang90** [comment]
@yichengq I am going to merge it since the build has passed.

**Comment by yichengq** [comment]
lgtm

**Pull request merged**